integrated_channels/integrated_channel/learner_data.py

```diff
@@ -97,28 +96,28 @@ def collect_learner_data(self):
 
         # Fetch course details from the Course API, and cache between calls.
         course_details = None
-        course_api = None
 
         for enterprise_enrollment in enrollment_queryset:
 
-            # Omit any enrollments where consent has not been granted
-            if not enterprise_enrollment.consent_available:
-                continue
-
             course_id = enterprise_enrollment.course_id
 
             # Fetch course details from Courses API
             if course_details is None or course_details['course_id'] != course_id:
-                if course_api is None:
-                    course_api = CourseApiClient()
-                course_details = course_api.get_course_details(course_id)
+                if self.course_api is None:
+                    self.course_api = CourseApiClient()
+                course_details = self.course_api.get_course_details(course_id)
 
             if course_details is None:
                 # Course not found, so we have nothing to report.
                 LOGGER.error("No course details found for enrollment %d: %s",
                              enterprise_enrollment.pk, course_id)
                 continue
 
+            # Since the course exists, see if the course enrollment's
+            # configuration may entail skipping data collection.
+            if self._should_skip_enrollment(enterprise_enrollment):
```

Comment:
Sort of: we're in the middle of looping through an enrollment queryset here, which is itself part of a function that indicates data collection (`collect_learner_data`). So I target the nearest scope with this wording (namely, the looping through the enrollment queryset), saying "skip this element". `_should_skip_data_collection` doesn't signify for _what_ we're skipping data collection and comes off too general.

We can compromise and make it `_should_skip_enrollments_data_collection`?